core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -44,7 +45,9 @@ object RequestChannel extends Logging {
     byteBuffer
   }
 
-  case class Request(processor: Int, connectionId: String, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
+  case class Session(principal: Principal, host: String)
+
+  case class Request(processor: Int, connectionId: String, session: Session = null, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
```

Comment:
Nope :) Actually, since we always have a Session, I don't think Option is needed. 
This was left from previous iteration.